### ISSUE TITLE
Trait Suggestion: Completing P201 and P202

### ISSUE THREAD

**Issue by david20000813** [pi-base/data#933]
## Trait Suggestion

The aim of this issue is to complete the list of spaces having [P201](https://topology.pi-base.org/properties/P000201) or not, and similarly for [P202](https://topology.pi-base.org/properties/P000202).

[S44](https://topology.pi-base.org/spaces/S000044) has P201: It is easy to see from definition that $\frac{1}{4}$ is contained in all nonempty open sets, whence it is a generic point.

[S46](https://topology.pi-base.org/spaces/S000046) does not have P201: The intersection of $S_n$ in the definition of the topology is easily seen to be empty. Since $S_n$ are all nonempty open sets, there is no point contained in all nonempty open sets, i.e., the space has no generic point.

[S48](https://topology.pi-base.org/spaces/S000048) has P201: The zero ideal, which is prime, is clearly contained in all $V_x$, whence it is generic.

[S200](https://topology.pi-base.org/spaces/S000200) does not have P201: The intersection of all $[n, \rightarrow)$ is clearly empty. Since $[n, \rightarrow)$ are all nonempty open sets, there is no point contained in all nonempty open sets, i.e., the space has no generic point.

[S118](https://topology.pi-base.org/spaces/S000118) does not have P202: Having a point with a unique neighborhood is equivalent to the union of all open proper subsets being not equal to the entire space. In this case, $\omega \times (\{0\} \cup \\{1/n\\}_{n=2}^\infty)$ and $\omega \times \{1\}$ are both open proper subsets, and the union of the two is the entire space. Hence, the space does not admit a point with a unique neighborhood.

**Comment by david20000813** [comment]
@GeoffreySangston I just went through the list of spaces for which either P201 or P202 is unknown, didn’t realize there’s already a PR about one. I agree that likely some of these should be possible to derive from stronger traits. I’ll check tomorrow if there are other unknown traits for these spaces that imply P201/P202 (or their negation). But I think it’s good to keep this list just in case nothing can be figured out.

**Comment by GeoffreySangston** [comment]
@david20000813 Also, what do think about contributing via git? There's some information here, and I found the video helpful.

https://github.com/pi-base/data/wiki/Contributing

(I'll leave it to the experienced contributors to help with the process, but I would definitely suggest making a very small and focused first PR.)

**Comment by david20000813** [comment]
@GeoffreySangston I’m learning the process right now. I’ll probably start with the two theorems I suggested for Čech-complete spaces.

**Comment by prabau** [comment]
@david20000813 What's your experience with git and with github?
Also, which editor are you using?

**Comment by david20000813** [comment]
@prabau I don’t have any experience with GitHub, which is why I didn’t try to contribute directly but instead asked you and Steven to add the theorem stating a path-connected space with a dispersion point has either a focal point or a generic point. I’m willing to learn though, especially now that I have multiple things I’d like to see added.

I’m not sure what you meant by “editor”? Are you referring to programming environment or something? If so, I have Visual Studio installed on my laptop.

**Comment by prabau** [comment]
Glad to have you on board.  Will give some suggestions tomorrow.

**Comment by prabau** [comment]
@david20000813
I assume the Visual Studio you have is vscode?  That's perfect.  It's the most appropriate editor for working on pi-base, as we use some special pi-base plugins for facilitating the work (more on that another time).  There is also an online version of vscode, called "github.dev" for doing the work directly from the browser.

Anyway, the first thing would be to get a high-level conceptual understanding of git as a source code control system.  There a lots of tutorials and videos on the web.  No need to get bogged down with too many details.  But you need to understand the notion of commit and of branch. (We use the "topic branch" workflow for pi-base, https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows) I recommend reading some of the documentation in https://docs.github.com/en  under "Get started" and going through the "Hello world" tutorial to create your first git repository.  Then play with it, do edits, commits, create or modify one or multiple files, create separate branches, merge them into your main branch, etc.

You can also look at the comments made to some of the other contributors here when they got onboarded.  Too much to repeat here, but there is a lot of useful info.  See in particular the parts having to do with github onboarding in:
#593
#677
#686

Once you feel comfortable with git/github, let us know and we can plan some really simple PRs to get the basic mechanism down. (I'd recommend something trivial first, not Cech-complete related).  Does not matter if these things get done tomorrow or in a week, or two weeks.  But feel free to ask anything at any time.

**Comment by david20000813** [comment]
While I’m still learning about PR’s, let me first report what happened with my search for stronger traits for these spaces:

1. S44: The strongest trait I can find that implies P201 is P192, quasi-sober;
2. S46: For this one, unfortunately it seems not having P201 is already the strongest assertion, given the currently known theorems. If it were not contractible (which is currently unknown to pi-Base), that would have been a stronger trait, but I figured out that it is actually contractible;
3. S48: This is already mentioned in the PR about S48. The strongest trait is that it is spectral;
4. S200: The strongest trait I can find that implies P201 is P56, meager;
5. S118: As it turns out, this one actually has P202. The definition in pi-Base misses one line from its definition in Counterexamples in Topology that clarifies the only neighborhood of the origin is the entire space. As is, the definition is sort of ambiguous as to what happens to the origin. As for the strongest trait, P202 itself or P146 both work. They imply each other under currently known traits and theorems.

**Comment by david20000813** [comment]
@prabau Hi, Patrick. I think I've more or less gotten comfortable with the process. I've made some edits on my personal fork of the project, test-adding the theorem that Cech-complete implies $k_3$. Since you seem to be suggesting something else for me to start with, I didn't submit a PR, but I think I've also learned the process as to how to do so.

**Comment by prabau** [comment]
Were you able to do the Hello world tutorial?  In particular, play with it and for example do a commit with multiple files?  Also create branches, merge branches, delete branches?  Doing all that in the Hello world repository is the best way to play with these things without impacting anything else.

Also note that we follow the "topic branch" workflow for pi-base.  I.e., it is recommended to have a separate branch (based on the `main` branch of your repository) for each set of changes in a PR.  That allows to have multiples independent changes going in parallel at the same time.  Otherwise, if you do everything from the main branch (not recommended), it becomes very hard to disentangle both sets of changes and kind of pain to wait for review of two PR at the same time, and if they don't get merged in the same order,... it becomes a mess.

What do you think?

**Comment by david20000813** [comment]
@prabau Done and done. I originally didn't put my Cech-complete changes in a separate branch, but I've now redone it. Is the suggestion to separate two somewhat connected theorems that I want to add to two different branches and two different PRs, or is it fine to do those in one branch and one PR? (I'm referring to #929 and #930.)

**Comment by prabau** [comment]
My comment was a general comment, not related to Cech-complete. In general, especially if there are unrelated changes that each may require more review or discussion, it's best to keep things separate. For Cech-complete, I don't think it matters.

In any case, can you humor me and in your Hello world repo, do one commit involving multiple files in the same commit and check there is no problem doing so?

After that, you can do a real PR for pi-base.  I would suggest not about Cech-complete.  You can pick something simple involving some traits for a space, together with maybe cleanup of some traits for the same space, or anything simple like that of your choice.  Just to check the github mechanics.

**Comment by david20000813** [comment]
@prabau Ah, I guess I got ahead of myself a bit. I already did a PR for #928. But yes, I just did a commit involving two files in my Hello world repository. There was no issue, as far as I can tell. (I've also already done that in my fork of pi-Base for Cech-complete theorems, without any issue as well.)

**Comment by prabau** [comment]
> @prabau Ah, I guess I got ahead of myself a bit. I already did a PR for #928. But yes, I just did a commit involving two files in my Hello world repository. There was no issue, as far as I can tell. (I've also already done that in my fork of pi-Base for Cech-complete theorems, without any issue as well.)

Perfect! Thank you.

**Comment by prabau** [comment]
#944 is nice, and not that we don't want to take your word for it, but if possible, we like to at least take a look at the referenced papers.

Would you mind doing a really trivial PR, adding or removing some simple trait somewhere, so we can go through the github motions?

**Comment by david20000813** [comment]
> #944 is nice, and not that we don't want to take your word for it, but if possible, we like to at least take a look at the referenced papers.

@prabau Sure, that is how the process is supposed to work, after all. As I noted in my comment in the PR, the journal website seems to be down at the moment, but the paper can still be accessed using Wayback Machine.

> Would you mind doing a really trivial PR, adding or removing some simple trait somewhere, so we can go through the github motions?

Sure. Maybe I can make a PR noting S46 does not have P201? The proof is just one line. (And, as I noted in a previous comment, there's no stronger trait for S46 that implies P201 under currently available theorems.) Or would you suggest something else?

**Comment by prabau** [comment]
Yeah, that's fine.

**Comment by prabau** [comment]
@david20000813 At this point, you can maybe stop writing new PRs for today.  The thing is that any PR that you currently write are from your own repository.  But these do not have access to the preview feature from the pi-base web site.  As there is no problem with your contributions, @StevenClontz will give you access to create branches directly from the official pi-base repository.  That will enable automatic builds and the preview feature.
Will make life easier for all.

**Comment by david20000813** [comment]
@prabau Oh, sure. I'll hold off on anything else to when I gain access then. Would it be preferred if, after I gain access, I remove the current PRs and copy-paste the files from my own repository into the official repository and resubmit PRs?

**Comment by prabau** [comment]
No need to resubmit anything.  We can keep going with the existing PRs.

**Comment by david20000813** [comment]
@prabau Just to check, if I don't see the "New branch" button in the all branches page, that means I have yet to gain access, is that correct? I'm just wondering when I will be able to proceed with the other PR's I want to work on and I'm not sure how to check.

**Comment by prabau** [comment]
We have to wait for @StevenClontz.  Not sure when he will see this, but once he gives you access, he will mention it to you.

**Comment by david20000813** [comment]
@prabau Understood. Thanks!

**Comment by david20000813** [comment]
Closed by various PRs
